### PR TITLE
[libfork] new port

### PR DESCRIPTION
**Pull request by ConorWilliams**
This PR adds [libfork](https://github.com/ConorWilliams/libfork), a [portable continuation stealing framework](https://arxiv.org/abs/2402.18480). I am the author of the package.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libfork/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(PACKAGE_NAME "libfork")
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
+
+file(INSTALL "${SOURCE_PATH}/LICENSE.md" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (jimwang118) [inline ports/libfork/portfile.cmake:19]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE.md")
```

File: ports/libfork/usage [hunk 2]
```diff
@@ -0,0 +1,4 @@
+libfork provides CMake targets:
+
+    find_package(libfork CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE libfork::libfork)
```
Comment (jimwang118) [inline ports/libfork/usage:4]: Add a blank line to the end of the file.

File: ports/libfork/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "libfork",
+  "version": "3.7.1",
```
Comment (jimwang118) [inline ports/libfork/vcpkg.json:3]:
```suggestion
  "version": "${VERSION}",
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

**Comment by jimwang118** [comment]
May I ask where the compiled lib file is installed?

**Comment by ConorWilliams** [comment]
> May I ask where the compiled lib file is installed?

It is a header-only lib

**Review by jimwang118 — DISMISSED** [review]

**Comment by jimwang118** [comment]
Usage test pass with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

**Comment by ConorWilliams** [comment]
> Usage test pass with following triplets:
> 
> ```
> x86-windows
> x64-windows
> x64-windows-static
> ```

Fantastic, are we good to merge?

**Pull request merged**